scripts/run-e2e-tests.sh

```diff
@@ -101,12 +101,15 @@ function install_kind() {
 }
 
 function install_spark() {
-  check_cmd "wget"
-  wget -qO-  https://archive.apache.org/dist/spark/spark-3.3.1/spark-3.3.1-bin-hadoop3.tgz | tar xzvf - && chmod +x spark-3.3.1-bin-hadoop3 && sudo mv spark-3.3.1-bin-hadoop3 /opt/.
-  exit_on_error "install spark failed."
-  sudo ln -s /opt/spark-3.3.1-bin-hadoop3 /opt/spark
-  exit_on_error "install spark failed. unable to create symlink"
-  export SPARK_HOME=/opt/spark/
+  PWD=$(pwd)
+  if [ ! -d "${PWD}/_spark" ]; then
+    echo  "Installing spark in yunikorn rep."
+    check_cmd "wget"
+    mkdir "${PWD}/_spark"
+    curl https://archive.apache.org/dist/spark/spark-3.3.1/spark-3.3.1-bin-hadoop3.tgz | tar xzvf - && chmod +x spark-3.3.1-bin-hadoop3 && mv spark-3.3.1-bin-hadoop3 "${PWD}/_spark/."
```

Comment:
Instead of moving, change into the _spark directory first -- this way the archive will go into the correct place and we don't litter up the top level directory if the script aborts.